src/store/auto_increment.rs

```diff
@@ -0,0 +1,19 @@
+use {
+    crate::result::{MutResult, Result},
+    async_trait::async_trait,
+};
+
+#[async_trait(?Send)]
+pub trait AutoIncrement
+where
+    Self: Sized,
+{
+    async fn get_increment_value(&self, table_name: &str, column_name: &str) -> Result<i64>;
+
+    async fn set_increment_value(
+        self,
+        table_name: &str,
+        column_name: &str,
+        end: i64,
+    ) -> MutResult<Self, ()>;
```

Comment:
In your previous commit, I remember you used `generate` and return `Range` type.
I thought the decision to use `Range` was really good.
Is there any special reason to divide `generate` one function into two?

It looks like that for `AutoIncrement` trait, it is enough to have 2 methods.

* `async fn generate_increment_values` which contains `get_increment_value` and `set_increment_values`

* `async fn update_increment_value` or maybe `values`? for manual update on auto increment column.

Reply:
I was working on a function which does it all efficienctly with one call but I couldn't get it to work and now my computer has been reset.

The reason I split it back up is for efficiency and niceness, how it is now is notably better.

I'll reply again in a minute with specifics.

Reply:
Oops! I Didn't end up replying.

The reason I moved away from range is that it is entirely unnessary, we know the end value will be start + length.

That said, not quite sure what my reasoning for moving away from the single function is.
Having two functions like this makes it so that if, in future, we want to be able set the value to a specific value we can.

That said, should we perhaps add a third which combines the two to simplify?

I was trying to merge them into one as well as making it so there's one function call for all columns which we can then put into a transaction but then lost that work.

I'll redo that.

Reply:
The issue, however, with doing it as the same function is that we can't undo the the updates if there's an error.
But oh well, not super important at them moment.